### PR TITLE
Drop EOLed php versions

### PR DESCRIPTION
**Pull request by simPod**
- add Github Actions to replace old CI (part 1/3)
- upgrade PHPUnit to v9
- lock symfony/var-dumper to v5.2 as @stable is too wide for testing (lowest is incompatible, symfony/var-dumper/Resources/functions/dump.php is missing)

### CODE REVIEW

**Comment by simPod** [comment]
@bpolaszek is there a way to enable Actions for this repo?

**Comment by bpolaszek** [comment]
Sure, you just need to push a `.github` directory with the workflow in it.

**Comment by bpolaszek** [comment]
> @bpolaszek is there a way to enable Actions for this repo?

If you want to test them, I think you can temporarily remove the `on.push.branches` config path, so that any push on any branch triggers the CI. Or, maybe adding `2.0.x-dev` in the `on.pull_request.branches` may work, but unsure.

**Comment by simPod** [comment]
I thought pull_request would be enough but maybe it's for master only 🤔  lemme see

**Comment by simPod** [comment]
That runs in my repo https://github.com/simPod/php-iterable-functions/actions/runs/488994536

Is there nothing to activate here https://github.com/bpolaszek/php-iterable-functions/actions for example?

**Comment by bpolaszek** [comment]
Theoretically, all I have to do is to merge your PR, actions cannot be triggered unless there's a workflow in the repository, AFAIK.

File: .github/workflows/continuous-integration.yml [hunk 1]
```diff
@@ -0,0 +1,76 @@
+name: "Continuous Integration"
+
+on:
+  pull_request:
+  push:
+  schedule:
+    - cron: "42 3 * * *"
+
+env:
+  fail-fast: true
```
Comment (bpolaszek) [inline .github/workflows/continuous-integration.yml]: I don't think it's necessary, is it?

File: .github/workflows/continuous-integration.yml [hunk 2]
```diff
@@ -0,0 +1,76 @@
+name: "Continuous Integration"
+
+on:
+  pull_request:
+  push:
+  schedule:
+    - cron: "42 3 * * *"
```
Comment (bpolaszek) [inline .github/workflows/continuous-integration.yml]: Didn't even know that feature! Why not just doing it `@monthly`?

File: .github/workflows/continuous-integration.yml [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+        with:
+          dependency-versions: "${{ matrix.dependencies }}"
+
+      - name: "Run PHPUnit"
+        run: "vendor/bin/phpunit --coverage-clover=coverage.xml"
+
+      - name: "Upload coverage file"
+        uses: "actions/upload-artifact@v2"
+        with:
+          name: "phpunit-sqlite-${{ matrix.deps }}-${{ matrix.php-version }}.coverage"
```
Comment (bpolaszek) [inline .github/workflows/continuous-integration.yml]: Why "sqlite"?

**Comment by bpolaszek** [comment]
For v2, I was also considering:
- Using [Pest](https://pestphp.com) for tests, as I've started doing this in some other [repositories](https://github.com/bpolaszek/picker/blob/2.0/tests/PickerTest.php). 
- Namespacing functions under `BenTools\IterableFunctions` instead of having them in the global namespace.

WDYT?

File: .github/workflows/continuous-integration.yml (hunk 2 above)
Comment (simPod) [reply .github/workflows/continuous-integration.yml]: Ide tells me 

image

can use it tho if you with

**Comment by simPod** [comment]
I don't know Pest so feel free to go on. I might learn something ;)

I'll add CS, PHPStan and psalm in next steps.

As for namespacing, yes why not!

File: .github/workflows/continuous-integration.yml (hunk 2 above)
Comment (bpolaszek) [reply .github/workflows/continuous-integration.yml]: try if this works, otherwise use `0 0 1 * *`.

Comment (simPod) [reply .github/workflows/continuous-integration.yml]: invalid `cron` attribute "@monthly" 

used `0 0 1 * *`

**Comment by simPod** [comment]
@bpolaszek let's ship this

**Comment by bpolaszek** [comment]
> @bpolaszek let's ship this

let's skip what?

**Comment by simPod** [comment]
ship 🚢  :shipit: 😄  merge

**Comment by bpolaszek** [comment]
LMAO 😄 That's what happens when I only had one liter of coffee on the Monday morning ☕

**Pull request merged**